aspnetcore/security/authorization/resourcebased.md

```diff
@@ -62,102 +56,50 @@ Task<bool> AuthorizeAsync(ClaimsPrincipal user,
-    }
-}
-```
+### Operational requirements
 
-You can see the handler works on `OperationAuthorizationRequirement`. The code inside the handler must take the Name property of the supplied requirement into account when making its evaluations.
+If you're making decisions based on the outcomes of CRUD (**C**reate, **R**ead, **U**pdate, **D**elete) operations, use the [OperationAuthorizationRequirement](/dotnet/api/microsoft.aspnetcore.authorization.infrastructure.operationauthorizationrequirement) helper class. This class enables you to write a single handler instead of an individual class for each operation type. To use it, provide some operation names:
 
-To call an operational resource handler you need to specify the operation when calling `AuthorizeAsync` in your action. For example:
+[!code-csharp[](resourcebased/samples/ResourceBasedAuthApp2/Services/DocumentAuthorizationCrudHandler.cs?name=snippet_OperationsClass)]
 
-```csharp
-if (await _authorizationService.AuthorizeAsync(User, document, Operations.Read))
-{
-    return View(document);
-}
-else
-{
-    return new ChallengeResult();
-}
-```
+The handler is implemented as follows, using an `OperationAuthorizationRequirement` requirement and a `Document` resource:
+
+[!code-csharp[](resourcebased/samples/ResourceBasedAuthApp2/Services/DocumentAuthorizationCrudHandler.cs?name=snippet_Handler)]
+
+The preceding handler validates the operation using the resource, the user's identity, and the requirement's `Name` property.
+
+To call an operational resource handler, specify the operation when invoking `AuthorizeAsync` in your action. For example:
+
+# [ASP.NET Core 2.x](#tab/aspnetcore2x)
+
+[!code-csharp[](resourcebased/samples/ResourceBasedAuthApp2/Controllers/DocumentController.cs?name=snippet_DocumentViewAction&highlight=11)]
+
+# [ASP.NET Core 1.x](#tab/aspnetcore1x)
+
+[!code-csharp[](resourcebased/samples/ResourceBasedAuthApp1/Controllers/DocumentController.cs?name=snippet_DocumentViewAction&highlight=11)]
+
+---
 
-This example checks if the User is able to perform the Read operation for the current `document` instance. If authorization succeeds the view for the document will be returned. If authorization fails returning `ChallengeResult` will inform any authentication middleware authorization has failed and the middleware can take the appropriate response, for example returning a 401 or 403 status code, or redirecting the user to a login page for interactive browser clients.
+The preceding example checks if the user is able to perform the read operation for the provided document. If authorization succeeds, the view for the document is returned. If authorization fails, returning `ChallengeResult` informs any authentication middleware that authorization failed, and the middleware can take the appropriate response. An appropriate response could be returning a 401 or 403 status code. For interactive browser clients, it could mean redirecting the user to a login page.
```

Comment:
Auto 403 was removed from 2.0. You now should call Forbid the the user is authenticated but not authorized.

Reply:
Or use the IPolicyEvaluator which basically does that (and is what MVC uses)